zipline/errors.py

```diff
@@ -259,7 +259,9 @@ class ConsumeAssetMetaDataError(ZiplineError):
     Raised when AssetMetaData.consume() is called on an invalid object.
     """
     msg = """
-AssetMetaData can not consume {obj}. MetaData must be a dict, a DataFrame, or
+AssetFinder can not consume metadata of type {obj}. Metadata must be a dict, a
+DataFrame, or an object with a 'read' attribute. If the provided metadata has a
```

Comment:
Lots of types have `read` methods (notably, any open file object).  I would refer to this specifically as a `tables.Table`, which is what we actually support.